### ISSUE TITLE
`ctx.query` doesn't work in some cases with `aot: true`

### ISSUE THREAD

**Issue by bogeychan** [elysiajs/elysia#656]
### What version of Elysia.JS is running?

1.0.22

### What platform is your computer?

WSL Ubuntu

### What steps can reproduce the bug?

```ts
import { Elysia, type Context } from 'elysia'

const checker = {
	check: async (ctx: Context, name: string, state?: string) => {
		return typeof state !== 'undefined'
	}
}

new Elysia()
	.derive((ctx) => {
		const { name } = ctx.params

		return {
			check: async () => {
				const { state } = ctx.query

				if (!(await checker.check(ctx, name, state ?? ctx.query.state))) {
					throw new Error('State mismatch')
				}
			}
		}
	})
	.get('/:name', async (ctx) => {
		await ctx.check()
		return 'yay'
	})
	.listen(8080)

```

Open http://localhost:8080/yay?state=123

### What is the expected behavior?

Respond with `yay` (no error thrown)

### What do you see instead?

```json
{"name":"Error","message":"State mismatch"}
```

### Additional information

AOT does not recognize:

- `const { state } = ctx.query`
- `ctx.query.state`
- `undefined ?? ctx.query.state`
- `const { query: { state } } = ctx`

only this seems to work:

`const state = ctx.query.state`

---

`const { state } = ctx.query` and `const { query: { state } } = ctx` used to work back in `1.0.3`:

- https://github.com/bogeychan/elysia-oauth2/blob/22e6c171eb5e80cc68c7ddfcb3811635c3d880ef/src/index.ts#L200
- https://github.com/bogeychan/elysia-oauth2/blob/22e6c171eb5e80cc68c7ddfcb3811635c3d880ef/src/index.ts#L241

**Comment by SaltyAom** [comment]
346134b published under 1.0.23, including test case. Let me know if it works on your end.

![doro running](https://media.tenor.com/a0at-3dnDzkAAAAM/doro-dorothy.gif)

**Comment by bogeychan** [comment]
Works for me, thnx
![image](https://github.com/elysiajs/elysia/assets/109291149/0055e245-be17-4440-8d8b-5dd71fe4ec4d)
